index.js

```diff
@@ -565,6 +565,9 @@ function Argv (processArgs, cwd) {
 
   function setPlaceholderKeys (argv) {
     Object.keys(options.key).forEach(function (key) {
+      // don't set placeholder keys for dot
+      // notation options 'foo.bar'.
+      if (key.match(/.*\..*/)) return
```

Comment:
I think you're right that we could probably simply opt for:

```js
~key.indexOf('.')
```

I was thinking about cases like `.foo` or `foo.`, but those are strange option names.